// gitbook-plugin-youtube/_assets/plugin.js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.GitBookPlugin = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
"use strict";

var GitBook = require('gitbook-core');
var React = GitBook.React;

/**
 * Block to render a youtube video.
 * @type {ReactClass}
 */

var YoutubeBlock = React.createClass({
    displayName: "YoutubeBlock",

    propTypes: {
        isVideo: React.PropTypes.bool.isRequired,
        url: React.PropTypes.string.isRequired,
        thumbnail: React.PropTypes.string.isRequired
    },

    render: function render() {
        var _props = this.props,
            isVideo = _props.isVideo,
            url = _props.url,
            thumbnail = _props.thumbnail;


        return React.createElement(
            "div",
            { className: "YoutubeBlock" },
            React.createElement(GitBook.ImportCSS, { href: "gitbook/youtube/youtube.css" }),
            isVideo ? React.createElement("iframe", { src: url,
                frameBorder: "0",
                allowFullScreen: true }) : React.createElement("img", { src: thumbnail })
        );
    }
});

module.exports = GitBook.createPlugin({
    activate: function activate(dispatch, getState, _ref) {
        var Components = _ref.Components;

        dispatch(Components.registerComponent(YoutubeBlock, { role: 'block:youtube' }));
    }
});

},{"gitbook-core":"gitbook-core"}]},{},[1])(1)
});
